### PR TITLE
Getting started with Constraint Layout design in Android

### LINKED ISSUE #1017: Getting started with Constraint view group design in Android
### Brief Summary: 

Over the past few years, Android has undergone a lot of improvements.  As an android developer, learning newly released technologies enable you to create powerful, lightweight, and modern applications. Constraint view group, also called constraint view alignment, is one of the newest skills that you need to arrange views and widgets on the screen the best way.

### Key Takeaways: 

By the end of this tutorial the learner should be able to;

1. Implement the concepts of chained alignment of views in a view group
2. Create a parent-child view group relationship
3. Design a responsive user interface - one that fits all devices screen sizes 
4. Generate a data binding object for referencing views within the view group

### References: 

N/A

### PR DESCRIPTION
**Pull request by Ericgacoki**
#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.

**NOTE: (1st time contributors - submit ONLY one article to get familiar with the process and style-once published, you can then submit any number of articles.)**
      
#### Grammar Checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [ ] Have you added sources for quotes and images that aren't yours?

#### Technical Checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution Guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**

Fixes #1017

### CODE REVIEW

**Comment by Ericgacoki** [comment]
@kayere, could you please let me know how's the progress ?

File: articles/getting-started-with-constraint-layout-in-android/index.md [hunk 1]
```diff
@@ -0,0 +1,570 @@
...
+</androidx.constraintlayout.widget.ConstraintLayout>
+
+```
+Views and widgets are added inside the layout tag which then becomes the parent to the respective element. In the above code, the device's screen serves as the layout's parent.
+
+### Adding a view into a Constraint layout
+
+Views and widgets are those components that the app users see on the screen. We can drag and drop the desired element from the `palette` onto the design preview. Alternatively, we can add the respective element in the XML code.
+
+Each element has four "circular-shaped-dots" on its edges known as `Constraint handles`. They're used to constrain/attach/align a view to a certain position on the screen. Unconstrained views float to the top of the screen hence it is important to give at least three constraints. In the later part of the tutorial, we'll learn about cases when we don't need to constrain a view at all. Meanwhile, a newly added view (in this case a `Button`) looks like this.
```
Comment (kayere) [inline articles/getting-started-with-constraint-layout-in-android/index.md:48]: It is atleast two constraints. One vertical and one horizontal constraint.

File: articles/getting-started-with-constraint-layout-in-android/index.md [hunk 2]
```diff
@@ -0,0 +1,570 @@
...
+        android:text="C"
+        app:layout_constraintBottom_toBottomOf="parent"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintHorizontal_bias="0.5"
+        app:layout_constraintStart_toEndOf="@+id/b"
+        app:layout_constraintTop_toTopOf="parent" />
+
+```
+
+For a chain to exist between two views, there must be a 1:1 alignment ratio between them.
```
Comment (kayere) [inline articles/getting-started-with-constraint-layout-in-android/index.md:343]: Please explain what 1:1 alignment ratio is.

**Review by kayere — COMMENTED** [review]
Hello @Ericgacoki, great article. I am done with the preliminary review and I have these few suggestions.

**Comment by Ericgacoki** [comment]
Hello @kayere, I'm done with the suggested changes. Please review. Thanks.

**Review by kayere — APPROVED** [review]

**Comment by hectorkambow** [comment]
plagiarism check done

**Comment by hectorkambow** [comment]
please go over the minor edits I made for the format @Ericgacoki  

cc @kayere

**Pull request merged**